store/postgres/src/store.rs

```diff
@@ -765,6 +668,128 @@ impl StoreTrait for Store {
+            )).filter(subgraph_deployments::node_id.eq(node_id.to_string()))
+            .load::<(String, String)>(&*self.conn.lock().unwrap())
+            .map_err(Error::from)
+            .map(|rows| {
+                rows.into_iter()
+                    .map(|(name, subgraph_id)| {
+                        let name = SubgraphDeploymentName::new(name)
+                            .expect("invalid subgraph name found in database");
+                        let subgraph_id = SubgraphId::new(subgraph_id)
+                            .expect("invalid subgraph ID found in database");
+                        (name, subgraph_id)
+                    }).collect()
+            })
+    }
+
+    fn write(
+        &self,
+        name: SubgraphDeploymentName,
+        subgraph_id: SubgraphId,
+        node_id: NodeId,
+    ) -> Result<(), Error> {
+        use db_schema::subgraph_deployments;
+
+        insert_into(subgraph_deployments::table)
+            .values((
+                subgraph_deployments::deployment_name.eq(name.to_string()),
+                subgraph_deployments::subgraph_id.eq(subgraph_id.to_string()),
+                subgraph_deployments::node_id.eq(node_id.to_string()),
+            )).on_conflict(subgraph_deployments::deployment_name)
+            .do_update()
+            .set((
+                subgraph_deployments::subgraph_id.eq(subgraph_id.to_string()),
+                subgraph_deployments::node_id.eq(node_id.to_string()),
+            )).execute(&*self.conn.lock().unwrap())
+            .map_err(Error::from)
+            .map(|_| ())
+    }
+
+    fn read(&self, name: SubgraphDeploymentName) -> Result<Option<(SubgraphId, NodeId)>, Error> {
+        use db_schema::subgraph_deployments;
+
+        subgraph_deployments::table
+            .select((
+                subgraph_deployments::subgraph_id,
+                subgraph_deployments::node_id,
+            )).filter(subgraph_deployments::deployment_name.eq(name.to_string()))
+            .first::<(String, String)>(&*self.conn.lock().unwrap())
+            .optional()
+            .map_err(Error::from)
+            .map(|row_opt| {
+                row_opt.map(|(subgraph_id, node_id)| {
+                    let subgraph_id = SubgraphId::new(subgraph_id)
+                        .expect("invalid subgraph ID found in database");
+                    let node_id = NodeId::new(node_id).expect("invalid node ID found in database");
+                    (subgraph_id, node_id)
+                })
+            })
+    }
+
+    fn remove(&self, name: SubgraphDeploymentName) -> Result<bool, Error> {
+        use db_schema::subgraph_deployments;
+
+        delete(subgraph_deployments::table)
+            .filter(subgraph_deployments::deployment_name.eq(name.to_string()))
+            .execute(&*self.conn.lock().unwrap())
+            .map(|row_count| match row_count {
+                0 => false,
+                1 => true,
+                _ => unreachable!(),
```

Comment:
See if `delete(subgraph_deployments::table.find(some_struct))` or something like that works, using `find` will make it locally clear that it's a primary key so this branch is unreachable.

Reply:
huh, I'm getting
```
no method named `find` found for type `diesel::query_builder::DeleteStatement<db_schema::subgraph_deployments::table, diesel::query_builder::where_clause::NoWhereClause>`
```

when I change filter to find. Might not be a thing for deletes?

Reply:
My suggestion was to put the `find` inside the delete. Don't know if it will work though.

Reply:
Oh nice, that works!